azure/services/privatedns/client_records.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package privatedns
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	azureautorest "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
+)
+
+// azureRecordsClient contains the Azure go-sdk Client for record sets.
+type azureRecordsClient struct {
+	recordsets privatedns.RecordSetsClient
+}
+
+// newRecordSetsClient creates a new record sets client from subscription ID.
+func newRecordSetsClient(auth azure.Authorizer) *azureRecordsClient {
+	recordsClient := privatedns.NewRecordSetsClientWithBaseURI(auth.BaseURI(), auth.SubscriptionID())
+	azure.SetAutoRestClientDefaults(&recordsClient.Client, auth.Authorizer())
+	return &azureRecordsClient{
+		recordsets: recordsClient,
+	}
+}
+
+// CreateOrUpdateAsync creates or updates a record asynchronously.
+// Well, I lied, CreateOrUpdateAsync creates a record synchronously. This is because azure api for records does not futures yet.
```

Comment:
```suggestion
// Well, I lied, CreateOrUpdateAsync creates a record synchronously. This is because azure api for records does not have futures yet.
```
Also side note I think we keep the same comment for `CreateOrUpdateAsync()` for all the services even if they're not actually returning a future.

Reply:
Actually I can do something like [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/a5505c9aec4bfa0bab54c8f078f5dbde7d404a1c/azure/services/securitygroups/client.go#L76) to make this async, let me try that.